src/coreclr/jit/lower.cpp

```diff
@@ -175,31 +176,36 @@ bool Lowering::IsSafeToContainMem(GenTree* parentNode, GenTree* childNode) const
 }
 
 //------------------------------------------------------------------------
-// IsSafeToContainMem: Checks for conflicts between childNode and grandParentNode
-// and returns 'true' iff memory operand childNode can be contained in ancestorNode
+// IsInvariantInRange: Check if a node is invariant in the specified range,
+// ignoring conflicts with one particular node.
 //
 // Arguments:
-//    grandParentNode - any non-leaf node
-//    parentNode - parent of `childNode` and an input to `grandParentNode`
-//    childNode  - some node that is an input to `parentNode`
+//    node         - The node.
+//    endExclusive - The exclusive end of the range to check invariance for.
+//    ignoreNode   - A node to ignore interference checks with, for example
+//                   because it will retain its relative order with 'node'.
 //
-// Return value:
-//    true if it is safe to make childNode a contained memory operand.
+// Returns:
+//    True if 'node' can be evaluated at any point between its current location
+//    and 'endExclusive' without giving a different result; otherwise false.
 //
-bool Lowering::IsSafeToContainMem(GenTree* grandparentNode, GenTree* parentNode, GenTree* childNode) const
+bool Lowering::IsInvariantInRange(GenTree* node, GenTree* endExclusive, GenTree* ignoreNode) const
 {
+    assert((node != nullptr) && (endExclusive != nullptr));
+
```

Comment:
Is there any invariant case we should handle here? That is, the prior handles the linear order being `node, endExclusive`. Should this handle `node, ignoreNode, endExclusive`?